logstash-core/lib/logstash/codecs/base.rb

```diff
@@ -11,7 +11,7 @@ module LogStash::Codecs; class Base < LogStash::Plugin
 
   def initialize(params={})
     super
-    config_init(@params)
+    config_init(@allow_env, @params)
```

Comment:
Oh, got it.. feel a bit of hacky but we still need to clean all of this, thanks for clarifying it for me.